Assets/Scripts/FaceCamera.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FaceCamera : MonoBehaviour
+{
+    // Start is called before the first frame update
+    void Start()
+    {
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        transform.LookAt(Camera.main.transform);
```

Comment:
Should only need to do this if our origin position changes - calculate once at the start once everything is set up, then only change when needed.

Reply:
That looks good for the initial positioning, but once the sphere is moved, rotated, etc., we need to continue to have the label look at the camera to make sure they are readable on a 2D screen (otherwise they get turned around, rotated, flipped, etc. - I tried to mimic, for now, what Nathan did in the NetLogo app).  However, this might end up being too jittery/jarring in AR to constantly rotate the label.  Another option is to have two setups: a screen space UI component for 2D screens and a diagetic component that has an initial position rotated towards the user/headset and potentially rotates towards the user/headset to increase readability.  Not sure, to be honest, what the best approach is.

Reply:
If a user is simply rotating the sphere or looking around, the labels will still be facing the origin - the only times they need to be re-calculated is when your origin position changes, so when the celestial sphere is moved as a whole, or if you are room-scale and move your own origin. I'm a fan of being lighter on the updates where possible:

Is the label in the camera's viewport?
Is the user's current position different from their last known position?

We don't need this firing every update unless either of those conditions are hit.

Reply:
I believe we're mixing up frames of reference. This isn't a Hololens issue - currently if you render the constellations, the stars are rendered as quads that are set to look at the camera just once on startup. They remain looking at the camera if you rotate their parent container, and the UI labels should continue to do the same in that scenario. Does that clarify what I'm thinking? If the UI labels are children of the parent, they'll retain their local rotations relative to the parent, and only the parent will be rotated. OR the alternative is that the user's transform is rotated - same result. If the user's transform position changes, OR the star container transform position changes, then yes we would need a recalculation. Hope that helps!

Reply:
Having discussed this in detail, we're in agreement that this needs to be tested on the Hololens for a final decision on how to approach labels.